indico/modules/events/editing/client/js/components/Timeline/judgment/TagInput.jsx

```diff
@@ -0,0 +1,77 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Dropdown} from 'semantic-ui-react';
+import {FinalField} from 'indico/react/forms/fields';
+import {Translate} from 'indico/react/i18n';
+
+import './TagInput.module.scss';
+
+/**
+ * A dropdown which allows tags to be searched and added
+ */
+function TagInput({onChange, value: defaultValue, placeholder, options}) {
+  return (
+    <Dropdown
+      placeholder={placeholder}
+      styleName="tag-input"
+      upward={false}
+      fluid
+      multiple
+      search
+      selection
+      defaultValue={defaultValue}
+      options={options.map(({id, verboseTitle, ...rest}) => ({
+        value: id,
+        text: verboseTitle,
+        key: id,
+        ...rest,
+      }))}
+      onChange={(_, {value: v}) => onChange(v)}
+      closeOnChange
+      renderLabel={({color, text}) => ({
+        color,
+        content: text,
+      })}
+    />
+  );
+}
+
+TagInput.propTypes = {
+  /** Called whenever the value changes (receives array of Strings) */
+  onChange: PropTypes.func.isRequired,
+  /** The default (initial) value */
+  value: PropTypes.arrayOf(PropTypes.number),
+  /** The text placeholder which should should up in the input box */
+  placeholder: PropTypes.string,
+  /** The list of options to show as valid tags */
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      code: PropTypes.string.isRequired,
+      title: PropTypes.string.isRequired,
+      verboseTitle: PropTypes.string.isRequired,
+      color: PropTypes.string.isRequired,
+    })
+  ).isRequired,
+};
+
+TagInput.defaultProps = {
+  value: [],
+  placeholder: Translate.string('Tag ID or title...'),
+};
+
+/** final-form-compatible <TagInput /> */
+export default function FinalTagInput({name, ...rest}) {
+  return <FinalField name={name} component={TagInput} {...rest} />;
+}
+
+FinalTagInput.propTypes = {
+  name: PropTypes.string.isRequired,
```

Comment:
options should be here as well since that's always required

Reply:
Hm... true. ESLint doesn't complain about it because we're just using `...rest`.